WeatherStationClient.py

```diff
@@ -0,0 +1,258 @@
+    
+    async def station_last_events(self, station_id, amount, sort=None):
+        """Read last X amount of station events. Optionally you can also sort them ASC or DESC."""
+        uri = f'station/{station_id}/events/last/{amount}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_events_between(self, station_id, from_unix_timestamp, to_unix_timestamp, sort=None):
+        """Read station events between time period you select. Optionally you can also sort them ASC or DESC."""
+        uri = f'station/{station_id}/events/from/{from_unix_timestamp}/to/{to_unix_timestamp}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+        
+    #TODO: Czy poprawna składnia URI jeśli nie ma filter
+    async def station_transmission_history_last(self, station_id, amount, filter_=None, sort=None):
+        """Read last X amount of station transmission history. Optionally you can also sort them ASC or DESC and filter."""
+        uri = f'station/{station_id}/history'
+        if filter_ is not None:
+            uri = uri + f'/{filter_}'
+        uri = uri + f'/{last}/{amount}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_transmission_history_between(self, station_id, from_unix_timestamp, to_unix_timestamp, filter_=None, sort=None):
+        """Read transmission history for specific time period. Optionally you can also sort them ASC or DESC and filter."""
+        uri = f'station/{station_id}/history/'
+        if filter_ is not None:
+            uri = uri + f'/{filter_}'
+        uri = uri + f'/from/{from_unix_timestamp}/to/{to_unix_timestamp}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_licenses(self, station_id):
+        """Retrieve all the licenses of your device. They are separated by the service (models, forecast ...)."""
+        return await self._send('GET', f'station/{station_id}/licenses')
+    
+    
+
+class ClientBuilder:
+
+    class _RequestContents:
+        def __init__(self, method, route, body, headers):
+            self.method = method
+            self.route = route
+            self.headers = headers
+            self.body = body
+    
+    class _ConnectionBase:
+        apiURI = 'https://api.fieldclimate.com/v1'
+    
+        async def __aenter__(self):
+            self._session = aiohttp.ClientSession()
+            return ApiClient(self)
+        
+        async def __aexit__(self, exc_type, exc_value, traceback):
+            await self._session.close()
+            
+        @abstractmethod
+        def _modifyRequest(self, requestContents):
+            pass
+        
+        async def _make_request(self, method, route, body=None):
```

Comment:
Should name be preceded with "_"? We use this method outside in api.

Reply:
The reason I preceded this name with `_` is that the user of this lib is not supposed to call this method. This was my key of using `_`'s - if the user is supposed to see / use / call something it should not be preceded with _ - otherwise it should.

If this goes against established conventions, by all means let's fix it :)

Reply:
... it would seem "_" is for internal while "__" is for private : https://docs.python.org/3.7/tutorial/classes.html#private-variables

so AFAIK my naming was correct in this case